internal/util/osio_token.go

```diff
@@ -0,0 +1,135 @@
+package util
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/satori/go.uuid"
+)
+
+type User struct {
+	Data Data `json:"data"`
+}
+
+type Data struct {
+	Attributes Attributes `json:"attributes"`
+}
+
+type Attributes struct {
+	Username string `json:"username"`
+	Email    string `json:"email"`
+	FullName string `json:"fullName"`
+}
+
+type Result struct {
+	Tokens []TokenResult `json:"tokens"`
+}
+
+type TokenResult struct {
+	UUID  string `json:"uuid"`
+	Token string `json:"token"`
+}
+
+type Token struct {
+	Data Data `json:"data"`
+}
+
+func CreateOSIOToken(env string, uuid string, key string, session string) (string, error) {
+	privateKey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(key))
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to parse private key file: %s", err))
+	}
+
+	user, err := loadUser(env, uuid)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to GET user for uuid '%s': %s", uuid, err))
+	}
+
+	token, err := generateToken(env, privateKey, user, uuid, session)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to generate token for uuid '%s': %s", uuid, err))
+	}
+
+	return token, nil
+}
+
+func generateToken(env string, key *rsa.PrivateKey, user User, userID string, session string) (string, error) {
+	token := jwt.New(jwt.SigningMethodRS256)
+	token.Header["kid"] = "0lL0vXs9YRVqZMowyw8uNLR_yr0iFaozdQk9rzq2OVU"
+
+	token.Claims.(jwt.MapClaims)["jti"] = uuid.NewV4().String()
+	token.Claims.(jwt.MapClaims)["exp"] = 1515627761
+	token.Claims.(jwt.MapClaims)["nbf"] = 0
+	token.Claims.(jwt.MapClaims)["iat"] = 1513035761
+	token.Claims.(jwt.MapClaims)["iss"] = "https://sso.openshift.io/auth/realms/fabric8"
+	token.Claims.(jwt.MapClaims)["aud"] = "fabric8-online-platform"
+	token.Claims.(jwt.MapClaims)["sub"] = userID
+	token.Claims.(jwt.MapClaims)["typ"] = "Bearer"
+	token.Claims.(jwt.MapClaims)["azp"] = "fabric8-online-platform"
+	token.Claims.(jwt.MapClaims)["auth_time"] = 1513035761
+	token.Claims.(jwt.MapClaims)["session_state"] = session
+	token.Claims.(jwt.MapClaims)["acr"] = "1"
+	token.Claims.(jwt.MapClaims)["approved"] = "true"
+	token.Claims.(jwt.MapClaims)["name"] = user.Data.Attributes.FullName
+	token.Claims.(jwt.MapClaims)["company"] = ""
+	token.Claims.(jwt.MapClaims)["preferred_username"] = user.Data.Attributes.Username
+	token.Claims.(jwt.MapClaims)["given_name"] = ""
+	token.Claims.(jwt.MapClaims)["family_name"] = ""
+	token.Claims.(jwt.MapClaims)["email"] = user.Data.Attributes.Email
+
+	//	token.Claims.(jwt.MapClaims)["iat"] = time.Now().Unix()
+	token.Claims.(jwt.MapClaims)["allowed-origins"] = []string{
+		environments[env].osioUrl,
```

Comment:
@alexeykazakov compared to your original code I changed these _allowed-origins_ URLs and adjusted them to match prod-preview. Other than that the method is pretty much unchanged. Does something else need to be adjusted or should I have kept the prod URLs?